File: c/raindrops/src/raindrops.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>
#include "raindrops.h"

char* convert(char result[], int drops) {

    /* Should use 'strcat_s' so we will have to assume char array is large enough */

    _Bool matched = false;

    if (drops % 3 == 0) {
        strcat(result, "Pling");
        matched = true;
    }

    if (drops % 5 == 0) {
        strcat(result, "Plang");
        matched = true;
    }

    if (drops % 7 == 0) {
        strcat(result, "Plong");
        matched = true;
    }

    /* No match so return int back as char array */
    if (!matched) {
        sprintf(result, "%d", drops);
    }

    return result;
}

File: ruby/raindrops/raindrops.rb
# RainDrops implementation
class Raindrops
  def self.convert(value)
    answer = ''

    answer << 'Pling' if (value % 3).zero?
    answer << 'Plang' if (value % 5).zero?
    answer << 'Plong' if (value % 7).zero?

    answer << value.to_s if answer.empty?

    answer
  end
end
